packages/nft/src/NFTHandler.ts

```diff
@@ -117,4 +120,68 @@ export class NFTHandler {
       tokenIds.map(tokenId => this.get(nftManager, storage, tokenId, nftFormat))
     )
   }
+
+  /**
+   * It takes a tokenId, and returns a transaction object
+   * @param {NFT721Manager} nftManager - NFT721Manager - The NFT721Manager instance that will be used
+   * to set the NFT tokenURI.
+   * @param {StorageType} storage - StorageType - The storage to save the NFT metadata.
+   * @param {NFT_METADATA_FORMAT} nftFormat - The format of the NFT metadata.
+   * @param {NFTData} nftData - The data of the NFT you want to set.
+   * @returns The transaction object
+   */
+  static async change(
+    nftManager: NFT721Manager,
+    storage: StorageType,
+    nftFormat: NFT_METADATA_FORMAT,
+    nftData: NFTData
+  ): Promise<interfaces.TransactionResponseExtended | undefined> {
+    const { tokenId, image, nftMetadata } = nftData
+
+    if (!nftManager.isActive) {
+      throw MUST_ACTIVATE
+    }
+
+    if (!nftManager.implements('setTokenURI', ['uint256', 'string'])) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      throw SET_TOKEN_URI_NOT_IMPLEMENTED(nftManager.contractAddr!)
+    }
+
+    let tokenUri = ''
+    switch (nftFormat) {
+      case NFT_METADATA_FORMAT.IMAGE:
+        if (!image) throw MISSING_PARAM('image (encoded as base64)')
+
+        tokenUri = await storage.storeBase64Image(image)
+        break
+
+      case NFT_METADATA_FORMAT.JSON:
+        if (!nftMetadata) throw MISSING_PARAM('nftMetadata')
+
+        tokenUri = await storage.storeMetadata(nftMetadata)
+        break
+
+      case NFT_METADATA_FORMAT.JSON_WITH_IMAGE: {
+        if (!image || !nftMetadata)
+          throw MISSING_PARAM('image (encoded as base64) or nftMetadata')
+
+        const imageUri = await storage.storeBase64Image(image)
+        nftMetadata['image'] = imageUri
+        tokenUri = await storage.storeMetadata(nftMetadata)
+        break
+      }
+    }
+
+    try {
+      const { transactionResponse } = await nftManager.execute({
+        method: 'setTokenURI(uint256,string)',
+        params: [tokenId, tokenUri]
+      })
+
+      return transactionResponse
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      throw TRANSACTION_FAILED(error)
+    }
+  }
```

Comment:
Why don't we just return what the `execute` returns? The `execute` function is already throwing this exception